src/lib/coda_networking/coda_networking.ml

```diff
@@ -519,23 +522,86 @@ module Make (Inputs : Inputs_intf) = struct
     let get_staged_ledger_aux_and_pending_coinbases_at_hash_rpc conn ~version:_
         hash =
       let hash_in_envelope = wrap_rpc_data_in_envelope conn hash in
-      get_staged_ledger_aux_and_pending_coinbases_at_hash hash_in_envelope
+      let%bind result =
+        get_staged_ledger_aux_and_pending_coinbases_at_hash hash_in_envelope
+      in
+      let%bind () =
+        if Option.is_none result then
+          Trust_system.(
+            record_envelope_sender config.trust_system config.logger
+              (Envelope.Incoming.sender hash_in_envelope)
+              Actions.
+                ( Requested_unknown_item
+                , Some
+                    ( "Requested staged ledge and pending coinbases at hash: \
+                       $hash"
+                    , [("hash", State_hash.to_yojson hash)] ) ))
+        else return ()
+      in
+      return result
     in
-    let answer_sync_ledger_query_rpc conn ~version:_ query =
-      let query_in_envelope = wrap_rpc_data_in_envelope conn query in
-      answer_sync_ledger_query query_in_envelope
+    let answer_sync_ledger_query_rpc conn ~version:_
+        ((hash, query) as sync_query) =
+      let sync_query_in_envelope = wrap_rpc_data_in_envelope conn sync_query in
+      let%bind result = answer_sync_ledger_query sync_query_in_envelope in
+      let%bind () =
+        match result with
```

Comment:
I don't like pattern matching on the `Or_error`, if we change `answer_sync_ledger_query` in the future to return errors in other situations this code will assume all errors are because the ledger was unknown.